### PR TITLE
[ZEPPELIN-1796] Add 'copy to clipboard' next to the paragraph id

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### What is this PR for?
There are lots of use cases that need to know each `Paragraph Id` in Zeppelin. 
e.g. [run paragraph using rest api](https://zeppelin.apache.org/docs/0.7.0-SNAPSHOT/rest-api/rest-notebook.html#run-a-paragraph-asynchronously) / [`z.angularBind`](https://zeppelin.apache.org/docs/0.7.0-SNAPSHOT/displaysystem/front-end-angular.html#basic-usage) / [export paragraphs using iframe](https://zeppelin.apache.org/docs/0.7.0-SNAPSHOT/manual/publish.html) ...

So I added `Copy to clipboard` btn next the each `Paragraph Id` under paragraph setting menu using [angular-zeroclipboard](https://github.com/lisposter/angular-zeroclipboard). 
It'll be useful even if it's small feature :)

### What type of PR is it?
Improvement | Feature 

### What is the Jira issue?
[ZEPPELIN-1796](https://issues.apache.org/jira/browse/ZEPPELIN-1796)

### How should this be tested?
1. `npm run build` under `zeppelin-web/`
2. `npm run start` to start dev mode and browse `http://localhost:9000`
3. click gear icon placed top of the each paragraph -> hover the mouse on the paragraph id 
4. click it and paste the paragraph id to anywhere to check it's copied well or not

### Screenshots (if appropriate)
copy_pid

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no 
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-web/src/app/app.js [hunk 1]
```diff
@@ -97,6 +98,10 @@
             combineDuplications: true,
             timeout: 6000
           });
+
+          uiZeroclipConfigProvider.setZcConf({
+            swfPath: '../bower_components/zeroclipboard/dist/ZeroClipboard.swf'
```
Comment (felixcheung) [inline zeppelin-web/src/app/app.js]: i thought we had a PR with this long time back. is this loading Flash?

**Comment by AhyoungRyu** [comment]
@felixcheung 
I just tested this patch only in dev mode(`npm run start`). But after I checked at `localhost:8080` and found   
```
404 http://localhost:8080/bower_components/zeroclipboard/dist/ZeroClipboard.swf
```
So I fixed it in here: [616b743](https://github.com/apache/zeppelin/pull/1750/commits/616b743019fe10c45a4693565e1d2c18dcd060fb). There won't be `ZeroClipboard.swf` loading issue anymore.

**Comment by felixcheung** [comment]
Great but the concerns are with dependencies on Flash with this approach

**Comment by AhyoungRyu** [comment]
@felixcheung Yeah you're right. So I changed the library from `angular-zeroclipboard` to [ngclipboard](https://github.com/sachinchoolur/ngclipboard/) which is not using flash. It says 
>Angularjs directive for [clipboard.js](https://clipboardjs.com/). A modern approach to copy text to clipboard without using flash.

And I checked it works well in `Chrome` / `Firefox` / `Safari`. 
Thanks for letting me know :)

**Comment by AhyoungRyu** [comment]
For the consistency with other menus in the dropdown menu, I removed the clippy icon and make paragraph id text itself as a button.
screen shot 2016-12-15 at 2 56 43 pmscreen shot 2016-12-15 at 2 56 51 pm

**Comment by AhyoungRyu** [comment]
CI has passed now. Ready to review again :)

**Comment by felixcheung** [comment]
LGTM!
Maybe we can leverage this for more copy to clipboard features?

**Comment by AhyoungRyu** [comment]
@felixcheung Yeah that's my plan :D

**Comment by AhyoungRyu** [comment]
Will merge into master if there are no more comments then

**Pull request closed without merging**